airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py

```diff
@@ -153,7 +153,7 @@ def read_records(
     @backoff.on_exception(
         backoff.expo,
         ReportGenerationFailure,
-        max_tries=5,
+        max_tries=1000,
```

Comment:
These value increasing is not ideal (5 to 1000 and 30 seconds to 120 seconds). It's possible to find an optimal way to handle those situations? @misteryeo this is the case where we should add options to users have a better control in connector config?

Reply:
It will be better if we can give these values user-defined @marcosmarxm

Reply:
Hi @marcosmarxm @harshithmullapudi ,
We have added an optional parameter for the user to enter report_wait_timeout. By default, it's 30 minutes, but if the user enters any time greater than 30 minutes, then that timing will be used.
Regarding max_tries, the majority of our time, we get reports at around 1000 retries. Thus we proposed to increase the value by 1000.
Since we are currently using this connector in our production, we request you to kindly merge this PR as soon as possible.